文章详情页白屏

### 是什么版本出现了此问题？

 1.5.0

### 在线站点地址

https://www.cenyuio.com/

### 发生了什么？

点击文章列表进入详情时卡顿，并打开文章详情后白屏

<img width="1821" height="846" alt="Image" src="https://github.com/user-attachments/assets/7e45b7b3-ad78-45e1-a08a-dcd6bbed3c72" /> 日志如下：2026-03-18T23:06:19.233+08:00 ERROR 7 --- [or-http-epoll-3] o.s.w.s.adapter.HttpWebHandlerAdapter    : [d3ff74ad-26563] Error [org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating SpringEL expression: "theme.config.custom.enable_edit and #authentication.name == post.owner.name" (template: "post" - line 31, col 7)] for HTTP GET "/archives/1773835131417", but ServerHttpResponse already committed (200 OK)

2026-03-18T23:06:19.234+08:00 ERROR 7 --- [or-http-epoll-3] r.n.http.server.HttpServerOperations     : [d3ff74ad-1, L:/172.18.0.15:8090 - R:/192.168.100.20:41756] Error finishing response. Closing connection



org.thymeleaf.exceptions.TemplateProcessingException: Exception evaluating SpringEL expression: "theme.config.custom.enable_edit and #authentication.name == post.owner.name" (template: "post" - line 31, col 7)

        at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:292) ~[thymeleaf-spring6-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        Suppressed: The stacktrace has been enhanced by Reactor, refer to additional information below: 

Error has been observed at the following site(s):

        *__checkpoint ⇢ Handler run.halo.app.theme.router.factories.PostRouteFactory$$Lambda/0x0000000027b18168@b7d02393 [DispatcherHandler]

        *__checkpoint ⇢ run.halo.app.security.InitializeRedirectionWebFilter@33d0cf10

        *__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]

        *__checkpoint ⇢ SwitchUserWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ AuthorizationWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ ExceptionTranslationWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ LogoutWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ ServerRequestCacheWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ SecurityContextServerWebExchangeWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]

        *__checkpoint ⇢ AnonymousAuthenticationWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]

        *__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]

        *__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ MapOAuth2AuthenticationFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]

        *__checkpoint ⇢  [DefaultWebFilterChain]

        *__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]

        *__checkpoint ⇢ DeviceSessionFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ ReactorContextWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ CsrfWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ CorsWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ HttpHeaderWriterWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]

        *__checkpoint ⇢ ServerWebExchangeReactorContextWebFilter [DefaultWebFilterChain]

        *__checkpoint ⇢ org.springframework.security.web.server.WebFilterChainProxy@f7023962

        *__checkpoint ⇢ run.halo.moments.rss.OldRssRouteRedirectionFilter@f3d8265d

        *__checkpoint ⇢ run.halo.comment.widget.captcha.CommentCaptchaFilter@58f6a4da

        *__checkpoint ⇢ run.halo.app.infra.webfilter.AdditionalWebFilterChainProxy@aed935a0

        *__checkpoint ⇢ org.springframework.web.filter.reactive.ServerWebExchangeContextFilter@17d0f6bf

        *__checkpoint ⇢ run.halo.app.infra.webfilter.LocaleChangeWebFilter@a32acaf

        *__checkpoint ⇢ org.springframework.web.filter.reactive.UrlHandlerFilter@23a7a662

        *__checkpoint ⇢ run.halo.app.theme.UserLocaleRequestAttributeWriteFilter@5bbef289

        *__checkpoint ⇢ HTTP GET "/archives/1773835131417" [ExceptionHandlingWebHandler]

Original Stack Trace:

                at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:292) ~[thymeleaf-spring6-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at run.halo.app.theme.ReactiveSpelVariableExpressionEvaluator.evaluate(ReactiveSpelVariableExpressionEvaluator.java:39) ~[classes/:2.23.1]

                at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.standard.expression.Expression.execute(Expression.java:125) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.standard.processor.StandardIfTagProcessor.isVisible(StandardIfTagProcessor.java:59) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.standard.processor.AbstractStandardConditionalVisibilityTagProcessor.doProcess(AbstractStandardConditionalVisibilityTagProcessor.java:61) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1314) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.engine.Model.process(Model.java:300) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.engine.StandaloneElementTagModelProcessable.process(StandaloneElementTagModelProcessable.java:114) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.engine.ProcessorTemplateHandler.handlePending(ProcessorTemplateHandler.java:2054) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.engine.ThrottledTemplateProcessor.process(ThrottledTemplateProcessor.java:234) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.engine.ThrottledTemplateProcessor.process(ThrottledTemplateProcessor.java:204) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.spring6.SpringWebFluxTemplateEngine$StreamThrottledTemplateProcessor.process(SpringWebFluxTemplateEngine.java:739) ~[thymeleaf-spring6-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at org.thymeleaf.spring6.SpringWebFluxTemplateEngine.lambda$createChunkedStream$2(SpringWebFluxTemplateEngine.java:269) ~[thymeleaf-spring6-3.1.3.RELEASE.jar:3.1.3.RELEASE]

                at reactor.core.publisher.FluxGenerate$GenerateSubscription.slowPath(FluxGenerate.java:270) ~[reactor-core-3.8.3.jar:3.8.3]

                at reactor.core.publisher.FluxGenerate$GenerateSubscription.request(FluxGenerate.java:212) ~[reactor-core-3.8.3.jar:3.8.3]

                at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.request(FluxContextWrite.java:138) ~[reactor-core-3.8.3.jar:3.8.3]

                at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.request(FluxPeekFuseable.java:144) ~[reactor-core-3.8.3.jar:3.8.3]

                at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.lambda$requestUpstream$0(FluxSubscribeOn.java:138) ~[reactor-core-3.8.3.jar:3.8.3]

                at reactor.core.scheduler.BoundedElasticThreadPerTaskScheduler$SchedulerTask.run(BoundedElasticThreadPerTaskScheduler.java:1016) ~[reactor-core-3.8.3.jar:3.8.3]

                at java.base/java.lang.VirtualThread.run(Unknown Source) ~[na:na]

Caused by: org.springframework.expression.spel.SpelEvaluationException: EL1001E: Type conversion problem, cannot convert from null to boolean

        at org.springframework.expression.spel.ast.OpAnd.assertValueNotNull(OpAnd.java:72) ~[spring-expression-7.0.5.jar:7.0.5]

        at org.springframework.expression.spel.ast.OpAnd.getBooleanValue(OpAnd.java:60) ~[spring-expression-7.0.5.jar:7.0.5]

        at org.springframework.expression.spel.ast.OpAnd.getValueInternal(OpAnd.java:50) ~[spring-expression-7.0.5.jar:7.0.5]

        at org.springframework.expression.spel.ast.SpelNodeImpl.getValue(SpelNodeImpl.java:114) ~[spring-expression-7.0.5.jar:7.0.5]

        at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:330) ~[spring-expression-7.0.5.jar:7.0.5]

        at org.thymeleaf.spring6.expression.SPELVariableExpressionEvaluator.evaluate(SPELVariableExpressionEvaluator.java:265) ~[thymeleaf-spring6-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at run.halo.app.theme.ReactiveSpelVariableExpressionEvaluator.evaluate(ReactiveSpelVariableExpressionEvaluator.java:39) ~[classes/:2.23.1]

        at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.standard.expression.Expression.execute(Expression.java:125) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.standard.processor.StandardIfTagProcessor.isVisible(StandardIfTagProcessor.java:59) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.standard.processor.AbstractStandardConditionalVisibilityTagProcessor.doProcess(AbstractStandardConditionalVisibilityTagProcessor.java:61) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.processor.element.AbstractAttributeTagProcessor.doProcess(AbstractAttributeTagProcessor.java:74) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.processor.element.AbstractElementTagProcessor.process(AbstractElementTagProcessor.java:95) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.util.ProcessorConfigurationUtils$ElementTagProcessorWrapper.process(ProcessorConfigurationUtils.java:633) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.engine.ProcessorTemplateHandler.handleOpenElement(ProcessorTemplateHandler.java:1314) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.engine.OpenElementTag.beHandled(OpenElementTag.java:205) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.engine.Model.process(Model.java:300) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.engine.StandaloneElementTagModelProcessable.process(StandaloneElementTagModelProcessable.java:114) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.engine.ProcessorTemplateHandler.handlePending(ProcessorTemplateHandler.java:2054) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.engine.ThrottledTemplateProcessor.process(ThrottledTemplateProcessor.java:234) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.engine.ThrottledTemplateProcessor.process(ThrottledTemplateProcessor.java:204) ~[thymeleaf-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.spring6.SpringWebFluxTemplateEngine$StreamThrottledTemplateProcessor.process(SpringWebFluxTemplateEngine.java:739) ~[thymeleaf-spring6-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at org.thymeleaf.spring6.SpringWebFluxTemplateEngine.lambda$createChunkedStream$2(SpringWebFluxTemplateEngine.java:269) ~[thymeleaf-spring6-3.1.3.RELEASE.jar:3.1.3.RELEASE]

        at reactor.core.publisher.FluxGenerate$GenerateSubscription.slowPath(FluxGenerate.java:270) ~[reactor-core-3.8.3.jar:3.8.3]

        at reactor.core.publisher.FluxGenerate$GenerateSubscription.request(FluxGenerate.java:212) ~[reactor-core-3.8.3.jar:3.8.3]

        at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.request(FluxContextWrite.java:138) ~[reactor-core-3.8.3.jar:3.8.3]

        at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.request(FluxPeekFuseable.java:144) ~[reactor-core-3.8.3.jar:3.8.3]

        at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.lambda$requestUpstream$0(FluxSubscribeOn.java:138) ~[reactor-core-3.8.3.jar:3.8.3]

        at reactor.core.scheduler.BoundedElasticThreadPerTaskScheduler$SchedulerTask.run(BoundedElasticThreadPerTaskScheduler.java:1016) ~[reactor-core-3.8.3.jar:3.8.3]

        at java.base/java.lang.VirtualThread.run(Unknown Source) ~[na:na]



### 相关日志输出

```shell

```

### 附加信息

_No response_

--- a/templates/page.html
+++ b/templates/page.html
@@ -54,7 +54,7 @@
     <button
       id="edit-page-btn"
       class="edit-page-btn"
-      th:if="${theme.config.custom.enable_edit and #authentication.name == singlePage.owner.name}"
+      th:if="${theme.config.custom.enable_edit and #authentication != null and #authentication.authenticated and #authentication.name == singlePage.owner.name}"
       th:data-url="@{/console/single-pages/editor(name=${singlePage.metadata.name}, returnToView=true)}"
       title="编辑页面"
     >

--- a/templates/post.html
+++ b/templates/post.html
@@ -28,7 +28,7 @@
     <button
       id="edit-post-btn"
       class="edit-post-btn"
-      th:if="${theme.config.custom.enable_edit and #authentication.name == post.owner.name}"
+      th:if="${theme.config.custom.enable_edit and #authentication != null and #authentication.authenticated and #authentication.name == post.owner.name}"
       th:data-url="@{${'/console/posts/editor'}(name=${post.metadata.name}, returnToView=true)}"
       title="编辑文章"
     >

